### ISSUE TITLE
Coordinates in `.obsm['spatial']` in Tables getting out of sync after transformations.

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/spatialdata#123]
In the [design doc](https://github.com/scverse/spatialdata/blob/main/docs/design_doc.md) we mention the possibility to store the centroids for labels and the centroids for graphs in the associated table, storing them with the intrinsic coordinate system. These coordinates could get out of sync after applying some transformations (for instance a cropping). We could take care of them and update them when we apply a transformation.

**Comment by giovp** [comment]
agree, I think the cropping and subsetting that is done at the spatialdata level should handle that. If the element is cropped outside spatialdata then I don't think we can give much guarantee

**Comment by LucaMarconato** [comment]
I believe the best is to have a new points layer, for instance the one returned from the new `get_centroids()` API. In this way there is no problem of getting out-of-sync because points are supported by the coordinate systems mechanism.

We should remove the references to `obsm['spatial']` in the design doc.

**Comment by LucaMarconato** [comment]
Updating the design doc now and closing this.
